### PR TITLE
feat: theme and content restructure

### PR DESCRIPTION
**Pull request by hferentschik**
Based on pull request #2615

### CODE REVIEW

**Comment by sharepointoscar** [comment]
Working through the broken link list, it will take some time but we will get there...

**Comment by hferentschik** [comment]
@sharepointoscar yeah, only 4 errors left :-)

```
- ./tmp-website/about/concepts/features/index.html
  *  internally linking to /developing/create-spring/, which does not exist (line 0)
     <a href="/developing/create-spring/">create new Spring Boot applications</a>
- ./tmp-website/community/talksdemos/index.html
  *  internally linking to /docs/resources/demos-talks-posts/articles/, which does not exist (line 0)
     <a href="/docs/resources/demos-talks-posts/articles/">Articles</a>
- ./tmp-website/docs/guides/tutorials/serverless-apps/index.html
  *  internally linking to /docs/resources/demos-talks-posts/create_spring/, which does not exist (line 0)
     <a href="/docs/resources/demos-talks-posts/create_spring/">create a new spring boot application</a>
- ./tmp-website/docs/guides/using-**/creating/create-spring/index.html
  *  internally linking to /docs/resources/demos-talks-posts/create_spring/, which does not exist (line 0)
     <a href="/docs/resources/demos-talks-posts/create_spring/">demo of using the command: ** create spring</a>
htmlproofer 3.15.3 | Error:  HTML-Proofer found 4 failures!
```

A couple I would be able to fix myself, but this one - `/docs/resources/demos-talks-posts/articles` is interesting. Seems like `articles.md` got removed. I am not sure whether this was intentional or whether you meant to move it but accidentally deleted it.

**Comment by sharepointoscar** [comment]
> @sharepointoscar yeah, only 4 errors left :-)
> 
> ```
> - ./tmp-website/about/concepts/features/index.html
>   *  internally linking to /developing/create-spring/, which does not exist (line 0)
>      <a href="/developing/create-spring/">create new Spring Boot applications</a>
> - ./tmp-website/community/talksdemos/index.html
>   *  internally linking to /docs/resources/demos-talks-posts/articles/, which does not exist (line 0)
>      <a href="/docs/resources/demos-talks-posts/articles/">Articles</a>
> - ./tmp-website/docs/guides/tutorials/serverless-apps/index.html
>   *  internally linking to /docs/resources/demos-talks-posts/create_spring/, which does not exist (line 0)
>      <a href="/docs/resources/demos-talks-posts/create_spring/">create a new spring boot application</a>
> - ./tmp-website/docs/guides/using-**/creating/create-spring/index.html
>   *  internally linking to /docs/resources/demos-talks-posts/create_spring/, which does not exist (line 0)
>      <a href="/docs/resources/demos-talks-posts/create_spring/">demo of using the command: ** create spring</a>
> htmlproofer 3.15.3 | Error:  HTML-Proofer found 4 failures!
> ```
> 
> A couple I would be able to fix myself, but this one - `/docs/resources/demos-talks-posts/articles` is interesting. Seems like `articles.md` got removed. I am not sure whether this was intentional or whether you meant to move it but accidentally deleted it.

It should not have been deleted, I see it in the location as shown below.
image

**Comment by hferentschik** [comment]
@sharepointoscar do you try do move these files into `content/en/docs/resources`? In this case more than one file is missing, it just happens to be that only one got linked to and we noticed it now due to the failing link check. Check your .gitignore, `resources` is in the ignore list due to the top level generated directory `resources`. None of the files you show in the above screenshot are part of version control anymore (hence also the greyed out fond in your screenshot). You will need to rename this directory or change .gitignore. We probably only want to exclude the top level `resources`. A leading slash should do.

**Comment by sharepointoscar** [comment]
> @sharepointoscar do you try do move these files into `content/en/docs/resources`? In this case more than one file is missing, it just happens to be that only one got linked to and we noticed it now due to the failing link check. Check your .gitignore, `resources` is in the ignore list due to the top level generated directory `resources`. None of the files you show in the above screenshot are part of version control anymore (hence also the greyed out fond in your screenshot). You will need to rename this directory or change .gitignore. We probably only want to exclude the top level `resources`. A leading slash should do.

Ok cool I'll add the leading slash on the `.gitignore` entry and re-submit.

File: content/en/docs/getting-started/_index.html [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+      Now you are ready to install Jenkins X into the cluster.
+      To do that, you will use the <code>jx boot</code> command.
+      For more details around JX Boot refer to the <a href="/docs/getting-started/setup/boot">Run Boot</a> section.
+      In an empty directory run:
+    </p>
+
+{{<highlight bash>}}
+# the parameter here, is the requirements yaml file
+# which was saved in the same directory you executed terraform apply.
+jx boot -r <path-to-terraform-directory>/jx-requirements.yml
```
Comment (vfarcic) [inline content/en/docs/getting-started/_index.html:227]: Please don't couple Terraform and JX Boot. One is not the requirement for the other, at least not until Terraform supports all the platforms where JX can run (which will probably never happen). I think that references like `path-to-terraform-directory` should be removed (there can be a note that's the option), and that this page should be split. Terraform is a helper to create a cluster, not a requirements. People should be able to install JX no matter how they created a cluster.

Comment (ethangj) [reply content/en/docs/getting-started/_index.html:227]: Agreed with Viktor

**Comment by ethangj** [comment]
I am good with this change now, we'll do a cleanup round after

**Comment by jha-cloudbees** [comment]
Looking better, more streamlined nav. As @deanesmith mentioned in a call, the tabs for the codeblocks need better shading/delineation to tell them apart, but other than that, this initial push lgtm!

Cheers on hunting down and fixing all the broken links as well!

**Comment by hferentschik** [comment]
@sharepointoscar, @jha-cloudbees, @ethangj - I guess we reached an MVP here. I took the WIP state off. It is technically my pull request since I opened it. I cannot self-approve it. Does someone want to give it a proper lgtm?

**Comment by hferentschik** [comment]
@jha-cloudbees - in case you want this pull request to merge you will need to prepend your 'lgtm' with a '\'. Prow commands start with a '\'.

**Comment by hferentschik** [comment]
@ethangj you are not in the OWNERS file. We need to clean it up really. I don't think @jha-cloudbees is either really.

File: content/en/docs/getting-started/_index.html (hunk 1 above)
Comment (hferentschik) [reply content/en/docs/getting-started/_index.html:227]: This is just in the context of this getting started guide where there are three successive steps to get started using the Terraform modules.

I agree that Terraform is not a requirement per se and it should be made clear. I am also inclined to think now that generating the initial `jx-requirements.yml` is taking it a bit too far. The disadvantages might actually outweigh the benefits. See also https://github.com/jenkins-x/terraform-google-jx/issues/75

**Comment by hferentschik** [comment]
@sharepointoscar, @jha-cloudbees, @ethangj - Yeah :-)

**Pull request merged**